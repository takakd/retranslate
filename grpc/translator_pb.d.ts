import * as jspb from 'google-protobuf'



export class TranslateRequest extends jspb.Message {
  getText(): string;
  setText(value: string): TranslateRequest;

  getSrclang(): LangType;
  setSrclang(value: LangType): TranslateRequest;

  getTargetlang(): LangType;
  setTargetlang(value: LangType): TranslateRequest;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): TranslateRequest.AsObject;
  static toObject(includeInstance: boolean, msg: TranslateRequest): TranslateRequest.AsObject;
  static serializeBinaryToWriter(message: TranslateRequest, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): TranslateRequest;
  static deserializeBinaryFromReader(message: TranslateRequest, reader: jspb.BinaryReader): TranslateRequest;
}

export namespace TranslateRequest {
  export type AsObject = {
    text: string,
    srclang: LangType,
    targetlang: LangType,
  }
}

export class TranslateResponse extends jspb.Message {
  getText(): string;
  setText(value: string): TranslateResponse;

  getSrclang(): LangType;
  setSrclang(value: LangType): TranslateResponse;

  getTranslatedtextlistMap(): jspb.Map<string, TranslatedText>;
  clearTranslatedtextlistMap(): TranslateResponse;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): TranslateResponse.AsObject;
  static toObject(includeInstance: boolean, msg: TranslateResponse): TranslateResponse.AsObject;
  static serializeBinaryToWriter(message: TranslateResponse, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): TranslateResponse;
  static deserializeBinaryFromReader(message: TranslateResponse, reader: jspb.BinaryReader): TranslateResponse;
}

export namespace TranslateResponse {
  export type AsObject = {
    text: string,
    srclang: LangType,
    translatedtextlistMap: Array<[string, TranslatedText.AsObject]>,
  }
}

export class TranslatedText extends jspb.Message {
  getText(): string;
  setText(value: string): TranslatedText;

  getLang(): LangType;
  setLang(value: LangType): TranslatedText;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): TranslatedText.AsObject;
  static toObject(includeInstance: boolean, msg: TranslatedText): TranslatedText.AsObject;
  static serializeBinaryToWriter(message: TranslatedText, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): TranslatedText;
  static deserializeBinaryFromReader(message: TranslatedText, reader: jspb.BinaryReader): TranslatedText;
}

export namespace TranslatedText {
  export type AsObject = {
    text: string,
    lang: LangType,
  }
}

export enum LangType { 
  UNKOWN = 0,
  JP = 1,
  EN = 2,
}
