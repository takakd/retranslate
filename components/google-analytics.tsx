/**
 * Google Analytics snippet code
 */
export const GoogleAnalytics = () => {
  return (
    <template>
      {/*Global site tag (gtag.js) - Google Analytics*/}
      <script
        async
        src="https://www.googletagmanager.com/gtag/js?id=G-CPT8TKNZH7"
      ></script>
      <script
        dangerouslySetInnerHTML={{
          __html: `window.dataLayer = window.dataLayer || [];
            function gtag(){dataLayer.push(arguments);}
            gtag('js', new Date());
            gtag('config', 'G-CPT8TKNZH7');`,
        }}
      />
    </template>
  )
}
