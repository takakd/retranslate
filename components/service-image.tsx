import React from 'react'

/**
 * Translation API service name
 */
export enum Service {
  AWS = 'aws',
  Google = 'google',
}

/**
 * ServiceImage component props
 */
export class ServiceImageProps {
  /**
   * Translation API service name
   */
  service: Service
}

/**
 * ServiceImage component
 *
 * @param {ServiceImageProps} props
 */
export const ServiceImage = (props: ServiceImageProps) => {
  if (props.service === Service.Google) {
    return <img src="/logo-google.png" alt="google" width="20" height="20" />
  } else if (props.service === Service.AWS) {
    return <img src="/logo-aws.png" alt="aws" width="28" height="20" />
  }
}
