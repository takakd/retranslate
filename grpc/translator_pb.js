// source: translator.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf')
var goog = jspb
var global = Function('return this')()

goog.exportSymbol('proto.translator.LangType', null, global)
goog.exportSymbol('proto.translator.TranslateRequest', null, global)
goog.exportSymbol('proto.translator.TranslateResponse', null, global)
goog.exportSymbol('proto.translator.TranslatedText', null, global)
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.translator.TranslateRequest = function (opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null)
}
goog.inherits(proto.translator.TranslateRequest, jspb.Message)
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.translator.TranslateRequest.displayName =
    'proto.translator.TranslateRequest'
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.translator.TranslateResponse = function (opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null)
}
goog.inherits(proto.translator.TranslateResponse, jspb.Message)
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.translator.TranslateResponse.displayName =
    'proto.translator.TranslateResponse'
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.translator.TranslatedText = function (opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null)
}
goog.inherits(proto.translator.TranslatedText, jspb.Message)
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.translator.TranslatedText.displayName =
    'proto.translator.TranslatedText'
}

if (jspb.Message.GENERATE_TO_OBJECT) {
  /**
   * Creates an object representation of this proto.
   * Field names that are reserved in JavaScript and will be renamed to pb_name.
   * Optional fields that are not set will be set to undefined.
   * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
   * For the list of reserved names please see:
   *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
   * @param {boolean=} opt_includeInstance Deprecated. whether to include the
   *     JSPB instance for transitional soy proto support:
   *     http://goto/soy-param-migration
   * @return {!Object}
   */
  proto.translator.TranslateRequest.prototype.toObject = function (
    opt_includeInstance
  ) {
    return proto.translator.TranslateRequest.toObject(opt_includeInstance, this)
  }

  /**
   * Static version of the {@see toObject} method.
   * @param {boolean|undefined} includeInstance Deprecated. Whether to include
   *     the JSPB instance for transitional soy proto support:
   *     http://goto/soy-param-migration
   * @param {!proto.translator.TranslateRequest} msg The msg instance to transform.
   * @return {!Object}
   * @suppress {unusedLocalVariables} f is only used for nested messages
   */
  proto.translator.TranslateRequest.toObject = function (includeInstance, msg) {
    var f,
      obj = {
        text: jspb.Message.getFieldWithDefault(msg, 1, ''),
        srclang: jspb.Message.getFieldWithDefault(msg, 2, 0),
        targetlang: jspb.Message.getFieldWithDefault(msg, 3, 0),
      }

    if (includeInstance) {
      obj.$jspbMessageInstance = msg
    }
    return obj
  }
}

/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.translator.TranslateRequest}
 */
proto.translator.TranslateRequest.deserializeBinary = function (bytes) {
  var reader = new jspb.BinaryReader(bytes)
  var msg = new proto.translator.TranslateRequest()
  return proto.translator.TranslateRequest.deserializeBinaryFromReader(
    msg,
    reader
  )
}

/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.translator.TranslateRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.translator.TranslateRequest}
 */
proto.translator.TranslateRequest.deserializeBinaryFromReader = function (
  msg,
  reader
) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break
    }
    var field = reader.getFieldNumber()
    switch (field) {
      case 1:
        var value = /** @type {string} */ (reader.readString())
        msg.setText(value)
        break
      case 2:
        var value = /** @type {!proto.translator.LangType} */ (
          reader.readEnum()
        )
        msg.setSrclang(value)
        break
      case 3:
        var value = /** @type {!proto.translator.LangType} */ (
          reader.readEnum()
        )
        msg.setTargetlang(value)
        break
      default:
        reader.skipField()
        break
    }
  }
  return msg
}

/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.translator.TranslateRequest.prototype.serializeBinary = function () {
  var writer = new jspb.BinaryWriter()
  proto.translator.TranslateRequest.serializeBinaryToWriter(this, writer)
  return writer.getResultBuffer()
}

/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.translator.TranslateRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.translator.TranslateRequest.serializeBinaryToWriter = function (
  message,
  writer
) {
  var f = undefined
  f = message.getText()
  if (f.length > 0) {
    writer.writeString(1, f)
  }
  f = message.getSrclang()
  if (f !== 0.0) {
    writer.writeEnum(2, f)
  }
  f = message.getTargetlang()
  if (f !== 0.0) {
    writer.writeEnum(3, f)
  }
}

/**
 * optional string text = 1;
 * @return {string}
 */
proto.translator.TranslateRequest.prototype.getText = function () {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ''))
}

/**
 * @param {string} value
 * @return {!proto.translator.TranslateRequest} returns this
 */
proto.translator.TranslateRequest.prototype.setText = function (value) {
  return jspb.Message.setProto3StringField(this, 1, value)
}

/**
 * optional LangType srcLang = 2;
 * @return {!proto.translator.LangType}
 */
proto.translator.TranslateRequest.prototype.getSrclang = function () {
  return /** @type {!proto.translator.LangType} */ (
    jspb.Message.getFieldWithDefault(this, 2, 0)
  )
}

/**
 * @param {!proto.translator.LangType} value
 * @return {!proto.translator.TranslateRequest} returns this
 */
proto.translator.TranslateRequest.prototype.setSrclang = function (value) {
  return jspb.Message.setProto3EnumField(this, 2, value)
}

/**
 * optional LangType targetLang = 3;
 * @return {!proto.translator.LangType}
 */
proto.translator.TranslateRequest.prototype.getTargetlang = function () {
  return /** @type {!proto.translator.LangType} */ (
    jspb.Message.getFieldWithDefault(this, 3, 0)
  )
}

/**
 * @param {!proto.translator.LangType} value
 * @return {!proto.translator.TranslateRequest} returns this
 */
proto.translator.TranslateRequest.prototype.setTargetlang = function (value) {
  return jspb.Message.setProto3EnumField(this, 3, value)
}

if (jspb.Message.GENERATE_TO_OBJECT) {
  /**
   * Creates an object representation of this proto.
   * Field names that are reserved in JavaScript and will be renamed to pb_name.
   * Optional fields that are not set will be set to undefined.
   * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
   * For the list of reserved names please see:
   *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
   * @param {boolean=} opt_includeInstance Deprecated. whether to include the
   *     JSPB instance for transitional soy proto support:
   *     http://goto/soy-param-migration
   * @return {!Object}
   */
  proto.translator.TranslateResponse.prototype.toObject = function (
    opt_includeInstance
  ) {
    return proto.translator.TranslateResponse.toObject(
      opt_includeInstance,
      this
    )
  }

  /**
   * Static version of the {@see toObject} method.
   * @param {boolean|undefined} includeInstance Deprecated. Whether to include
   *     the JSPB instance for transitional soy proto support:
   *     http://goto/soy-param-migration
   * @param {!proto.translator.TranslateResponse} msg The msg instance to transform.
   * @return {!Object}
   * @suppress {unusedLocalVariables} f is only used for nested messages
   */
  proto.translator.TranslateResponse.toObject = function (
    includeInstance,
    msg
  ) {
    var f,
      obj = {
        text: jspb.Message.getFieldWithDefault(msg, 1, ''),
        srclang: jspb.Message.getFieldWithDefault(msg, 2, 0),
        translatedtextlistMap: (f = msg.getTranslatedtextlistMap())
          ? f.toObject(
              includeInstance,
              proto.translator.TranslatedText.toObject
            )
          : [],
      }

    if (includeInstance) {
      obj.$jspbMessageInstance = msg
    }
    return obj
  }
}

/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.translator.TranslateResponse}
 */
proto.translator.TranslateResponse.deserializeBinary = function (bytes) {
  var reader = new jspb.BinaryReader(bytes)
  var msg = new proto.translator.TranslateResponse()
  return proto.translator.TranslateResponse.deserializeBinaryFromReader(
    msg,
    reader
  )
}

/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.translator.TranslateResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.translator.TranslateResponse}
 */
proto.translator.TranslateResponse.deserializeBinaryFromReader = function (
  msg,
  reader
) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break
    }
    var field = reader.getFieldNumber()
    switch (field) {
      case 1:
        var value = /** @type {string} */ (reader.readString())
        msg.setText(value)
        break
      case 2:
        var value = /** @type {!proto.translator.LangType} */ (
          reader.readEnum()
        )
        msg.setSrclang(value)
        break
      case 3:
        var value = msg.getTranslatedtextlistMap()
        reader.readMessage(value, function (message, reader) {
          jspb.Map.deserializeBinary(
            message,
            reader,
            jspb.BinaryReader.prototype.readString,
            jspb.BinaryReader.prototype.readMessage,
            proto.translator.TranslatedText.deserializeBinaryFromReader,
            '',
            new proto.translator.TranslatedText()
          )
        })
        break
      default:
        reader.skipField()
        break
    }
  }
  return msg
}

/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.translator.TranslateResponse.prototype.serializeBinary = function () {
  var writer = new jspb.BinaryWriter()
  proto.translator.TranslateResponse.serializeBinaryToWriter(this, writer)
  return writer.getResultBuffer()
}

/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.translator.TranslateResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.translator.TranslateResponse.serializeBinaryToWriter = function (
  message,
  writer
) {
  var f = undefined
  f = message.getText()
  if (f.length > 0) {
    writer.writeString(1, f)
  }
  f = message.getSrclang()
  if (f !== 0.0) {
    writer.writeEnum(2, f)
  }
  f = message.getTranslatedtextlistMap(true)
  if (f && f.getLength() > 0) {
    f.serializeBinary(
      3,
      writer,
      jspb.BinaryWriter.prototype.writeString,
      jspb.BinaryWriter.prototype.writeMessage,
      proto.translator.TranslatedText.serializeBinaryToWriter
    )
  }
}

/**
 * optional string text = 1;
 * @return {string}
 */
proto.translator.TranslateResponse.prototype.getText = function () {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ''))
}

/**
 * @param {string} value
 * @return {!proto.translator.TranslateResponse} returns this
 */
proto.translator.TranslateResponse.prototype.setText = function (value) {
  return jspb.Message.setProto3StringField(this, 1, value)
}

/**
 * optional LangType srcLang = 2;
 * @return {!proto.translator.LangType}
 */
proto.translator.TranslateResponse.prototype.getSrclang = function () {
  return /** @type {!proto.translator.LangType} */ (
    jspb.Message.getFieldWithDefault(this, 2, 0)
  )
}

/**
 * @param {!proto.translator.LangType} value
 * @return {!proto.translator.TranslateResponse} returns this
 */
proto.translator.TranslateResponse.prototype.setSrclang = function (value) {
  return jspb.Message.setProto3EnumField(this, 2, value)
}

/**
 * map<string, TranslatedText> translatedTextList = 3;
 * @param {boolean=} opt_noLazyCreate Do not create the map if
 * empty, instead returning `undefined`
 * @return {!jspb.Map<string,!proto.translator.TranslatedText>}
 */
proto.translator.TranslateResponse.prototype.getTranslatedtextlistMap =
  function (opt_noLazyCreate) {
    return /** @type {!jspb.Map<string,!proto.translator.TranslatedText>} */ (
      jspb.Message.getMapField(
        this,
        3,
        opt_noLazyCreate,
        proto.translator.TranslatedText
      )
    )
  }

/**
 * Clears values from the map. The map will be non-null.
 * @return {!proto.translator.TranslateResponse} returns this
 */
proto.translator.TranslateResponse.prototype.clearTranslatedtextlistMap =
  function () {
    this.getTranslatedtextlistMap().clear()
    return this
  }

if (jspb.Message.GENERATE_TO_OBJECT) {
  /**
   * Creates an object representation of this proto.
   * Field names that are reserved in JavaScript and will be renamed to pb_name.
   * Optional fields that are not set will be set to undefined.
   * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
   * For the list of reserved names please see:
   *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
   * @param {boolean=} opt_includeInstance Deprecated. whether to include the
   *     JSPB instance for transitional soy proto support:
   *     http://goto/soy-param-migration
   * @return {!Object}
   */
  proto.translator.TranslatedText.prototype.toObject = function (
    opt_includeInstance
  ) {
    return proto.translator.TranslatedText.toObject(opt_includeInstance, this)
  }

  /**
   * Static version of the {@see toObject} method.
   * @param {boolean|undefined} includeInstance Deprecated. Whether to include
   *     the JSPB instance for transitional soy proto support:
   *     http://goto/soy-param-migration
   * @param {!proto.translator.TranslatedText} msg The msg instance to transform.
   * @return {!Object}
   * @suppress {unusedLocalVariables} f is only used for nested messages
   */
  proto.translator.TranslatedText.toObject = function (includeInstance, msg) {
    var f,
      obj = {
        text: jspb.Message.getFieldWithDefault(msg, 1, ''),
        lang: jspb.Message.getFieldWithDefault(msg, 2, 0),
      }

    if (includeInstance) {
      obj.$jspbMessageInstance = msg
    }
    return obj
  }
}

/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.translator.TranslatedText}
 */
proto.translator.TranslatedText.deserializeBinary = function (bytes) {
  var reader = new jspb.BinaryReader(bytes)
  var msg = new proto.translator.TranslatedText()
  return proto.translator.TranslatedText.deserializeBinaryFromReader(
    msg,
    reader
  )
}

/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.translator.TranslatedText} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.translator.TranslatedText}
 */
proto.translator.TranslatedText.deserializeBinaryFromReader = function (
  msg,
  reader
) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break
    }
    var field = reader.getFieldNumber()
    switch (field) {
      case 1:
        var value = /** @type {string} */ (reader.readString())
        msg.setText(value)
        break
      case 2:
        var value = /** @type {!proto.translator.LangType} */ (
          reader.readEnum()
        )
        msg.setLang(value)
        break
      default:
        reader.skipField()
        break
    }
  }
  return msg
}

/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.translator.TranslatedText.prototype.serializeBinary = function () {
  var writer = new jspb.BinaryWriter()
  proto.translator.TranslatedText.serializeBinaryToWriter(this, writer)
  return writer.getResultBuffer()
}

/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.translator.TranslatedText} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.translator.TranslatedText.serializeBinaryToWriter = function (
  message,
  writer
) {
  var f = undefined
  f = message.getText()
  if (f.length > 0) {
    writer.writeString(1, f)
  }
  f = message.getLang()
  if (f !== 0.0) {
    writer.writeEnum(2, f)
  }
}

/**
 * optional string text = 1;
 * @return {string}
 */
proto.translator.TranslatedText.prototype.getText = function () {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ''))
}

/**
 * @param {string} value
 * @return {!proto.translator.TranslatedText} returns this
 */
proto.translator.TranslatedText.prototype.setText = function (value) {
  return jspb.Message.setProto3StringField(this, 1, value)
}

/**
 * optional LangType lang = 2;
 * @return {!proto.translator.LangType}
 */
proto.translator.TranslatedText.prototype.getLang = function () {
  return /** @type {!proto.translator.LangType} */ (
    jspb.Message.getFieldWithDefault(this, 2, 0)
  )
}

/**
 * @param {!proto.translator.LangType} value
 * @return {!proto.translator.TranslatedText} returns this
 */
proto.translator.TranslatedText.prototype.setLang = function (value) {
  return jspb.Message.setProto3EnumField(this, 2, value)
}

/**
 * @enum {number}
 */
proto.translator.LangType = {
  UNKOWN: 0,
  JP: 1,
  EN: 2,
}

goog.object.extend(exports, proto.translator)
