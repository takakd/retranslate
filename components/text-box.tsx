import React, { FormEvent, useRef } from 'react'

import { CloseButton, Spinner } from 'react-bootstrap'
import { ArrowRight, Clipboard } from 'react-bootstrap-icons'

import { ServiceImage } from '../components/service-image'

/**
 * Translation service name
 */
export enum Service {
  AWS = 'aws',
  Google = 'google',
}

/**
 * TextBoxInputEvent represents an event object of TextBoxProps.onInput event.
 */
export interface TextBoxInputEvent {
  /**
   * Text in textarea
   */
  text: string
}

/**
 * Props of TextBox component
 */
export interface TextBoxProps {
  /**
   * Initial text in textarea
   */
  text: string
  /**
   * Translation service name
   */
  service?: Service
  /**
   * Input event handler.
   * @param {TextBoxInputEvent} event event object.
   */
  onInput?: (event: TextBoxInputEvent) => void
  /**
   * Clear text event handler.
   */
  onClearText?: () => void
  /**
   * Class name
   */
  className?: string
  /**
   * Class name prefix of child elements.
   */
  classNamePrefix?: string
  /**
   * Placeholder text in textarea.
   */
  placeHolder?: string

  /**
   * Show arrow on the left of textarea.
   */
  showArrow?: boolean
  /**
   * Show loading icon on the left of textarea.
   */
  loading?: boolean
}

/**
 * TextBox component.
 *
 * @param {TextBoxProps} props
 */
export const TextBox = (props: TextBoxProps) => {
  const textareaRef = useRef(null)
  const className = props?.className ? props.className : 'text-box'
  const classNamePrefix = props?.classNamePrefix
    ? props.classNamePrefix
    : 'text-box'

  /**
   * Fired on inputting texts in textarea.
   * @param {React.FormEvent} event
   */
  const onInput = (event: FormEvent) => {
    if (!props.onInput) {
      return
    }

    const target = event.target as HTMLTextAreaElement
    props.onInput({
      text: target.value,
    })
  }

  /**
   * Fired on clicking the clipboard button.
   */
  const onClickClipboard = () => {
    // Copy text in textarea to the OS clipboard.
    textareaRef.current.select()
    document.execCommand('copy')
  }

  return (
    <div className={`d-flex flex-row align-items-center w-100 ${className}`}>
      {
        // Show arrow icon.
        props.showArrow && !props.loading && (
          <div className="icon">
            <ArrowRight size={20} />
          </div>
        )
      }
      {
        // Show loading icon when loading.
        props.loading && (
          <div className="icon">
            <Spinner animation="border" size="sm" />
          </div>
        )
      }
      <div className={`d-flex flex-column ${classNamePrefix}-text-container`}>
        {
          <div className={`${className}-header`}>
            {
              // Show service icon
              props.service && <ServiceImage service={props.service} />
            }
          </div>
        }
        <div
          className={`position-relative w-100 ${classNamePrefix}-textarea-container`}
        >
          <textarea
            ref={textareaRef}
            className={`h-100 ${classNamePrefix}-textarea`}
            placeholder={props?.placeHolder}
            defaultValue={props.text}
            onInput={(event) => onInput(event)}
          />
          {
            // Show erase button when text is entered and the event handler exists.
            props.text !== '' && props?.onClearText && (
              <CloseButton
                className={`position-absolute close-button p-1 ${classNamePrefix}-close-button`}
                onClick={props?.onClearText}
              />
            )
          }
          {
            // Show clipboard button when text is entered.
            props.text !== '' && (
              <Clipboard
                className={`position-absolute ${classNamePrefix}-clipboard-button`}
                onClick={onClickClipboard}
              />
            )
          }
        </div>
      </div>

      <style jsx>{`
        .${className}-header {
          min-height: 30px;
        }

        .${classNamePrefix}-textarea {
          width: 100%;
          resize: none;
          padding: 0.5rem 0.5rem 36px 0.5rem;
        }
        .${classNamePrefix}-text-container,
          .${classNamePrefix}-textarea-container {
          flex: 1;
        }
        .icon {
          min-width: 30px;
          text-align: center;
        }
        ::placeholder {
          color: #aaa;
          opacity: 1;
        }
      `}</style>
      <style jsx global>{`
        .${classNamePrefix}-close-button {
          top: 1px;
          right: 6px;
        }
        .${classNamePrefix}-clipboard-button {
          cursor: pointer;
          bottom: 10px;
          right: 10px;
        }
      `}</style>
    </div>
  )
}
