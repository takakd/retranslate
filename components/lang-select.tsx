import React, { useEffect, useState } from 'react'

import { Dropdown, DropdownButton } from 'react-bootstrap'

import { LangType } from '../grpc/translator_pb'

/**
 * LangSelectChangeEvent is the event object of LangSelectProps.onLangChange event.
 */
export interface LangSelectChangeEvent {
  /**
   * Selected language type
   */
  lang: LangType
}

/**
 * LangSelect component props
 */
export interface LangSelectProps {
  /**
   * Language type
   */
  lang?: LangType

  /**
   * LangType change event handler.
   * @param {LangSelectChangeEvent} event event object.
   */
  onLangChange?: (event: LangSelectChangeEvent) => void

  /**
   * Language options
   */
  langOptions: {
    label: string
    value: LangType
  }[]
}

/**
 * LangSelect component
 *
 * @param {LangSelectProps} props
 */
export const LangSelect = (props: LangSelectProps) => {
  const [langLabel, setLangLabel] = useState<string>('')

  useEffect(() => {
    // Set option label by props.lang.
    setLangLabel(toLangLabel(props.lang))
  })

  /**
   * Creates the language label by LangType.
   * @param {LangType} lang language type.
   * @returns {string} language label.
   */
  const toLangLabel = (lang: LangType): string => {
    for (const option of props.langOptions) {
      if (lang === option.value) {
        return option.label
      }
    }
    return 'Language'
  }

  /**
   * Fired on clicking the language option.
   * @param {LangType} lang Selected language.
   */
  const onClick = (lang: LangType) => {
    if (!props.onLangChange) {
      return
    }
    props?.onLangChange({
      lang,
    })
  }

  return (
    <DropdownButton
      key={props.lang}
      title={langLabel}
      variant="secondary"
      renderMenuOnMount={true}
    >
      {props.langOptions.map((option) => {
        return (
          <Dropdown.Item
            as="button"
            key={option.value}
            onClick={() => onClick(option.value)}
          >
            {option.label}
          </Dropdown.Item>
        )
      })}
    </DropdownButton>
  )
}
