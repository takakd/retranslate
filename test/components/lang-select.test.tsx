import React from 'react'

import { LangSelect, LangSelectProps } from '../../components/lang-select'
import { LangType } from '../../grpc/translator_pb'
import { fireEvent, render } from '../testUtils'

describe('LangSelect', () => {
  const langOptions = [
    {
      label: 'Japanese',
      value: LangType.JP,
    },
    {
      label: 'English',
      value: LangType.EN,
    },
  ]

  it('matches snapshot', () => {
    const props: LangSelectProps = {
      lang: LangType.JP,
      langOptions,
    }

    const { asFragment } = render(<LangSelect {...props} />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('language label', () => {
    const props: LangSelectProps = {
      lang: LangType.EN,
      langOptions,
    }

    const { container } = render(<LangSelect {...props} />)

    const labelButton = container.getElementsByClassName('dropdown-toggle')
    expect(labelButton[0].textContent).toBe('English')

    const buttons = container.getElementsByClassName('dropdown-item')
    props.langOptions.forEach((prop, i) => {
      expect(buttons[i].textContent).toBe(prop.label)
    })
  })

  it('on click', () => {
    const props: LangSelectProps = {
      lang: LangType.EN,
      langOptions,
      onLangChange: jest.fn(),
    }

    const { container } = render(<LangSelect {...props} />)

    const buttons = container.getElementsByClassName('dropdown-item')

    // Click Japanese
    fireEvent.click(buttons[0])
    expect(props.onLangChange).toBeCalledTimes(1)
    expect(props.onLangChange).toBeCalledWith({ lang: LangType.JP })
  })
})
