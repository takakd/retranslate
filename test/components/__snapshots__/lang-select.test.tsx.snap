// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LangSelect matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="dropdown"
  >
    <button
      aria-expanded="false"
      aria-haspopup="true"
      class="dropdown-toggle btn btn-secondary"
      type="button"
    >
      Japanese
    </button>
    <div
      aria-labelledby=""
      class="dropdown-menu"
      style="position: absolute; top: 0px; left: 0px; opacity: 0; pointer-events: none; margin: 0px;"
      x-placement="bottom-start"
    >
      <button
        class="dropdown-item"
      >
        Japanese
      </button>
      <button
        class="dropdown-item"
      >
        English
      </button>
    </div>
  </div>
</DocumentFragment>
`;
